packages/textinput/src/react/index.js

```diff
@@ -54,25 +49,36 @@ const styles = {
 const TextInput = (props, context) => {
   const themeName = context.themeName || themeDefaultName
   const allProps = { ...props, themeName }
-  const { error, label, icon, style, subLabel, ...inputProps } = allProps
+  const {
+    error,
+    fieldAfter,
+    label,
+    icon,
+    style,
+    subLabel,
+    ...inputProps
+  } = allProps
 
   return (
     <label
-      {...styles.input(allProps)}
+      {...styles.textInput(allProps)}
       {...(allProps.style ? { style: allProps.style } : null)}
       {...(allProps.className ? { className: allProps.className } : null)}
     >
       {label && <div {...styles.label(allProps)}>{label}</div>}
 
       <div {...styles.fieldContainer(allProps)}>
         <Halo error={error} gapSize={Halo.gapSizes.small}>
-          <input
-            {...filterReactProps(inputProps, { tagName: 'input' })}
-            {...styles.field(allProps)}
-            disabled={allProps.disabled}
-            placeholder={allProps.placeholder}
-            ref={allProps.innerRef}
-          />
+          <div {...styles.field(allProps)}>
+            <input
+              {...filterReactProps(inputProps, { tagName: 'input' })}
+              {...styles.fieldInput(allProps)}
+              disabled={allProps.disabled}
+              placeholder={allProps.placeholder}
+              ref={allProps.innerRef}
+            />
+            {fieldAfter && fieldAfter}
```

Comment:
I don't think the combo logic here is required, just `{fieldAfter}`.

```suggestion
            {fieldAfter}
```